### ISSUE TITLE
[ISSUE] Provider bug

### ISSUE THREAD

**Issue by grgouveia-everis** [databrickslabs/terraform-provider-databricks#433]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Databricks Provider issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

### Affected Resource(s)
Please list the resources as a list, for example:
- databricks.permissions

I believe the issue is related to associating this permission to the cluster because it takes a long to try to modify the cluster permissions but it doesn't manage to.

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl

resource "databricks_cluster_policy" "group_cluster_policies" {
  for_each   = var.v_cluster_policy_overrides
  name       = format("%s-cluster-policy", each.key)
  definition = jsonencode(merge(var.v_base_cluster_policy, each.value))

  depends_on = [databricks_group_member.dbks_grp_member]
}

resource "databricks_permissions" "can_use_cluster_policy" {
  for_each          = databricks_cluster_policy.group_cluster_policies
  cluster_policy_id = each.value.id
  access_control {
    group_name       = each.key
    permission_level = "CAN_USE"
  }
}


resource "databricks_cluster" "dbks_cluster" {
  for_each = var.v_clusters_configs

  spark_version           = each.value.spark_version
  cluster_name            = each.value.cluster_name
  autotermination_minutes = each.value.autotermination_minutes
  driver_node_type_id     = each.value.node_type_id
  node_type_id            = each.value.node_type_id
  policy_id               = databricks_cluster_policy.group_cluster_policies[each.key].id

  spark_conf = {
    "spark.databricks.delta.preview.enabled" = "true"
    "spark.speculation"                      = "true"
  }

  autoscale {
    min_workers = each.value.min_workers
    max_workers = each.value.max_workers
  }

  depends_on = [time_sleep.dbks_sp_ra, databricks_cluster_policy.group_cluster_policies]
}


```

### Debug Output
Please add turn on logging, e.g. `TF_LOG=DEBUG terraform apply` and run command again, paste it to gist & provide the link to gist. If you're still willing to paste in log output, make sure you provide only relevant log lines with requests.

It would make it more readable, if you pipe the log through `| grep databricks | sed -E 's/^.* plugin[^:]+: (.*)$/\1/'`, e.g.:

```
TF_LOG=DEBUG terraform plan 2>&1 | grep databricks | sed -E 's/^.* plugin[^:]+: (.*)$/\1/'
```

### Panic Output
https://gist.github.com/grgouveia-everis/6cb9114d56d6562ece4c25ff0d4fdd77
### Expected Behavior
Should've created the policy and associated to cluster.

### Actual Behavior
Creates policy, associate to cluster but panics out and in the next apply try to create the resource again. 

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Are there anything atypical about your accounts that we should know?

**Comment by nfx** [comment]
@grgouveia-everis please remove explicit assignment to admin group, because admin group has permissions to use anything anyway. this should fix it.

**Comment by grgouveia-everis** [comment]
Hi Serge, 
This reference to admin actually is a AAD group called admins I created at our infrastructure, it has no relations with the databricks admins group.

Anyway, I fixed it by downgrading the provider to 0.2.5, so I think it can be an issue with the version or maybe that the cluster was created with a version and the permission with another... don't know
